sdv/constraints/tabular.py

```diff
@@ -938,20 +918,40 @@ def _transform(self, table_data):
                 Transformed data.
         """
         low = table_data[self.low_column_name]
+        middle = table_data[self.middle_column_name]
         high = table_data[self.high_column_name]
 
-        data = logit(table_data[self.middle_column_name], low, high)
-        table_data[self._transformed_column] = data
-        table_data = table_data.drop(self.middle_column_name, axis=1)
+        if self._is_datetime:
+            diff_column_1 = get_datetime_diff(middle, low, self._dtype)
+            diff_column_2 = get_datetime_diff(high, middle, self._dtype)
+        else:
+            diff_column_1 = middle - low
+            diff_column_2 = high - middle
 
-        return table_data
+        table_data[self.diff_column_name_1] = np.log(diff_column_1 + 1)
+        table_data[self.diff_column_name_2] = np.log(diff_column_2 + 1)
+
+        list_columns_nans = [self.low_column_name, self.middle_column_name, self.high_column_name]
+        if add_nans_column(table_data, list_columns_nans):
```

Comment:
similarly to the other PR, I think we can just have this method return the nans column as a series and then do the rest if it is not None

Reply:
I agree, done in [18bdaf0](https://github.com/sdv-dev/SDV/pull/1404/commits/18bdaf09a65211edb2e8c54ccffd057a72fa68d4)